src/components/DataSubmissions/DeleteOrphanFileChip.tsx

```diff
@@ -0,0 +1,135 @@
+import { useMemo, useState } from "react";
+import { Chip, ChipProps, styled } from "@mui/material";
+import { useSnackbar } from "notistack";
+import { useMutation } from "@apollo/client";
+import { ReactComponent as DeleteFileIcon } from "../../assets/icons/delete_single_file_icon.svg";
+import { DELETE_ORPHANED_FILE, DeleteOrphanedFileResp } from "../../graphql";
+import { useAuthContext } from "../Contexts/AuthContext";
+
+const StyledChip = styled(Chip)({
+  "&.MuiChip-root": {
+    border: "1px solid #2B528B",
+    background: "#2B528B",
+    height: "23px",
+    marginLeft: "60.5px",
+    alignSelf: "center",
+  },
+  "& .MuiChip-icon": {
+    marginLeft: "3.62px",
+    marginRight: 0,
+    height: "100%",
+    width: "14.69px",
+    alignSelf: "center",
+    paddingTop: "2.65px",
+    paddingBottom: "3.66px",
+  },
+  "& .MuiChip-label": {
+    color: "#D8E3F2",
+    fontFamily: "'Inter', 'Rubik', sans-serif",
+    fontWeight: 400,
+    fontSize: "10px",
+    lineHeight: "12.1px",
+    paddingLeft: "3.66px",
+    paddingRight: "5px",
+    paddingTop: "2px",
+    paddingBottom: "3px",
+  },
+});
+
+/**
+ * The roles that are allowed to delete orphan files within a submission.
+ *
+ * @note The button is only visible to users with these roles.
+ */
+const DeleteOrphanFileRoles: User["role"][] = [
+  "Submitter",
+  "Organization Owner",
+  "Data Curator",
+  "Admin",
+];
+
+type Props = {
+  submission: Submission;
+  submittedID: QCResult["submittedID"];
+  onDeleteFile: (success: boolean) => void;
+} & ChipProps;
+
+const DeleteOrphanFileChip = ({
+  submission,
+  submittedID,
+  onDeleteFile,
+  disabled,
+  ...rest
+}: Props) => {
+  const { user } = useAuthContext();
+  const { enqueueSnackbar } = useSnackbar();
+  const [loading, setLoading] = useState<boolean>(false);
+  const isOrphanFile = useMemo(
+    () => submission?.fileErrors?.find((error) => error.submittedID === submittedID),
+    [submission?.fileErrors, submittedID]
+  );
+  const canDeleteOrphanedFiles = useMemo(() => {
+    if (
+      !user?.role ||
+      !DeleteOrphanFileRoles.includes(user.role) ||
+      !submission?.fileErrors?.length
+    ) {
+      return false;
+    }
+
+    return true;
+  }, [user, submission]);
+
+  const [deleteOrphanedFile] = useMutation<DeleteOrphanedFileResp>(DELETE_ORPHANED_FILE, {
+    context: { clientName: "backend" },
+    fetchPolicy: "no-cache",
+  });
+
+  const handleOnDelete = async () => {
+    setLoading(true);
+
+    try {
+      const { data: d, errors } = await deleteOrphanedFile({
+        variables: {
+          _id: submission._id,
+          fileName: submittedID,
+        },
+      });
+
+      if (errors || !d?.deleteOrphanedFile?.success) {
+        throw new Error("Unable to delete orphan file.");
+      }
```

Comment:
Since we don't enforce coverage on CRDC-DH, this is **COMPLETELY** optional–but this branch doesn't seem to be covered by the tests.

![Screenshot 2024-05-16 at 2 38 21 PM](https://github.com/CBIIT/crdc-datahub-ui/assets/38357871/c0f0b0e7-c668-404b-b7df-357944f17e62)

I don't know if it's the missing scenario for `success: false` or maybe missing GraphQL error scenario since Apollo has two different types of errors (network vs GraphQL level). e.g.

```javascript
  {
    request: {
      query: DELETE_ORPHANED_FILES,
      variables: { _id: "submission-id" },
    },
    result: {
      errors: [new GraphQLError("Mock GraphQL error")],
    },
  },
```

Reply:
Added both, but it seemed to be caused by the success false condition.